$NetBSD: patch-ac,v 1.1 2007/01/14 16:28:40 rillig Exp $

- A string<char> cannot be initialized with an unsigned char *.
- There is no constructor string(const char *, const char *).

--- zip.cc.orig	2004-11-07 21:14:28.000000000 +0100
+++ zip.cc	2007-01-14 17:11:15.104504656 +0100
@@ -452,7 +452,8 @@ void zip_entry::name_set(const string& A
 
 string zip_entry::name_get() const
 {
-	return string(file_name, file_name + info.filename_length);
+	/* file_name is an unsigned char *. */
+	return string(reinterpret_cast<const char *>(file_name), info.filename_length);
 }
 
 /** Check central directory entry. */
